api/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -124,6 +135,11 @@ public static long isoTimestamptzToMicros(String timestampString) {
         OffsetDateTime.parse(timestampString, DateTimeFormatter.ISO_DATE_TIME));
   }
 
+  public static long isoTimestampToNanos(CharSequence timestamp) {
+    return ChronoUnit.NANOS.between(
+        EPOCH, OffsetDateTime.parse(timestamp, DateTimeFormatter.ISO_DATE_TIME));
```

Comment:
Looks like this uses `ISO_DATE_TIME` format and `OffsetDateTime`, which allows a time zone offset and corresponds to `timestamptz` (see the micros equivalent immediately above). This should actually be `isoTimestamptzToNanos`.